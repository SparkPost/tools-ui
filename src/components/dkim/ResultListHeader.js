import React from 'react';
import { Link } from 'react-router';

import Icon from '../Icon';
import { CopyPopover } from '../popover/Popover';

// TODO Finish styling error
const Error = (props) => {
  const { error } = props;
  return (
    <div className='panel__body'>
      <Icon name='exclamation-circle' />
      {error.message}
    </div>
  );
};

const ResultListHeader = (props) => {
  const { email, getResults, error, loggedIn } = props;

  return (
    <div className='panel panel--accent'>
      <div className='panel__body'>

        <div className='float--right'>
          {!loggedIn && <Link className='actionLink' to='/'>Save Results</Link>}
          <CopyPopover>
            <a className='actionLink' title='Share'>Share</a>
          </CopyPopover>
          <a className='actionLink' onClick={getResults} title='Refresh Messages'>Refresh</a>
        </div>

        <p className='text--muted marginBottom--xs'>Generated Test Address</p>
        <h3>{email}</h3>

        {error && <Error error={error} />}
      </div>
    </div>
  );
};

ResultListHeader.defaultProps = {
  loggedIn: false
};

ResultListHeader.propTypes = {
  loggedIn: React.PropTypes.bool.isRequired
};

export default ResultListHeader;
