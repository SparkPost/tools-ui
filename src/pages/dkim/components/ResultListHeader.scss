@import "~@sparkpost/styles/src/assets/scss/shared/config";
@import "~@sparkpost/styles/src/assets/scss/shared/functions";

.dkimResultListHeader {
  &__label {
    color: grayscale(5);
    font-size: rem(14);

    @media screen and (min-width: break(md)) {
      font-size: rem(15);
    }
  }

  &__email {
    font-size: rem(15);
    line-height: rem(18);

    @media screen and (min-width: break(sm)) {
      font-size: rem(18);
      line-height: rem(24);
    }

    @media screen and (min-width: break(md)) {
      font-size: rem(21);
      line-height: rem(30);
    }
  }
}
