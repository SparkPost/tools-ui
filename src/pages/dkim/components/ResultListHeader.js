import React from 'react';
import { CopyPopover } from 'components/popover/Popover';
import { ActionLink, SaveResultsLink } from 'components/button/Button';
import config from 'config/index';

import './ResultListHeader.scss';

export default ({ email, getResults, loggedIn }) => (
  <div className='panel panel--accent'>
    <div className='panel__body'>

      <div className='float--right'>
        {!loggedIn && <SaveResultsLink extraQueryParams={config.queryParams.dkim}/>}
        <CopyPopover>
          <ActionLink title='Share' track={true}>Share</ActionLink>
        </CopyPopover>
        <ActionLink onClick={getResults} title='Refresh Messages' track={true}>Refresh</ActionLink>
      </div>

      <p className='dkimResultListHeader__label marginBottom--xs'>Generated Test Address</p>
      <h3 className='dkimResultListHeader__email'>{email}</h3>
    </div>
  </div>
);
