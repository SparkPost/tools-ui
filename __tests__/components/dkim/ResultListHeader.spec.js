import React from 'react';
import renderer from 'react-test-renderer';

import ResultListHeader from 'pages/dkim/components/ResultListHeader';

describe('ResultListHeader component', () => {

  const getResults = () => {}

  test('should render', () => {
    expect(renderer.create(<ResultListHeader getResults={getResults} email='some@email.com' />)).toMatchSnapshot();
  });

  test('should render with an error present', () => {
    expect(renderer.create(<ResultListHeader getResults={getResults} email='some@email.com' error={new Error('oops')} />)).toMatchSnapshot();
  })

  test('should render when logged in', () => {
    expect(renderer.create(<ResultListHeader getResults={getResults} email='some@email.com' loggedIn={true} />)).toMatchSnapshot();
  });

  test('should render when not logged in', () => {
    expect(renderer.create(<ResultListHeader getResults={getResults} email='some@email.com' loggedIn={false} />)).toMatchSnapshot();
  });

});
