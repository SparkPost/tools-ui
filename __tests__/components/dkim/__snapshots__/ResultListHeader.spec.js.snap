exports[`ResultListHeader component should render 1`] = `
<div
  className="panel panel--accent">
  <div
    className="panel__body">
    <div
      className="float--right">
      <span
        className="popover__group popover__hoverTrigger">
        <a
          className="actionLink"
          href="https://TEST.sparkpost.com/sign-up?return=http://localhost/"
          onClick={null}
          title="Save Results">
          Save Results
        </a>
        <span
          className="popover popover--top popover--m">
          Create a free SparkPost account or login into your account to save results
        </span>
      </span>
      <span
        className="popover__group"
        onClick={[Function]}>
        <a
          className="actionLink"
          onClick={[Function]}
          style={Object {}}
          title="Share">
          Share
        </a>
        <div
          className="popover copyPopover popover--l popover--rightAligned popover--bottom">
          <h6
            className="text--left copyPopover__label">
            <i
              className="fa fa-check-circle "
              style={
                Object {
                  "transition": "all 200ms ease-out",
                }
              } />
             Copied to Clipboard!
          </h6>
          <p
            className="copyPopover__copiedString">
            http://localhost/
          </p>
          <input
            className="copyPopover__hidden"
            readOnly={true}
            value="http://localhost/" />
        </div>
      </span>
      <a
        className="actionLink"
        onClick={[Function]}
        style={Object {}}
        title="Refresh Messages">
        Refresh
      </a>
    </div>
    <p
      className="dkimResultListHeader__label marginBottom--xs">
      Generated Test Address
    </p>
    <h3
      className="dkimResultListHeader__email">
      some@email.com
    </h3>
  </div>
</div>
`;

exports[`ResultListHeader component should render when logged in 1`] = `
<div
  className="panel panel--accent">
  <div
    className="panel__body">
    <div
      className="float--right">
      
      <span
        className="popover__group"
        onClick={[Function]}>
        <a
          className="actionLink"
          onClick={[Function]}
          style={Object {}}
          title="Share">
          Share
        </a>
        <div
          className="popover copyPopover popover--l popover--rightAligned popover--bottom">
          <h6
            className="text--left copyPopover__label">
            <i
              className="fa fa-check-circle "
              style={
                Object {
                  "transition": "all 200ms ease-out",
                }
              } />
             Copied to Clipboard!
          </h6>
          <p
            className="copyPopover__copiedString">
            http://localhost/
          </p>
          <input
            className="copyPopover__hidden"
            readOnly={true}
            value="http://localhost/" />
        </div>
      </span>
      <a
        className="actionLink"
        onClick={[Function]}
        style={Object {}}
        title="Refresh Messages">
        Refresh
      </a>
    </div>
    <p
      className="dkimResultListHeader__label marginBottom--xs">
      Generated Test Address
    </p>
    <h3
      className="dkimResultListHeader__email">
      some@email.com
    </h3>
  </div>
</div>
`;

exports[`ResultListHeader component should render when not logged in 1`] = `
<div
  className="panel panel--accent">
  <div
    className="panel__body">
    <div
      className="float--right">
      <span
        className="popover__group popover__hoverTrigger">
        <a
          className="actionLink"
          href="https://TEST.sparkpost.com/sign-up?return=http://localhost/"
          onClick={null}
          title="Save Results">
          Save Results
        </a>
        <span
          className="popover popover--top popover--m">
          Create a free SparkPost account or login into your account to save results
        </span>
      </span>
      <span
        className="popover__group"
        onClick={[Function]}>
        <a
          className="actionLink"
          onClick={[Function]}
          style={Object {}}
          title="Share">
          Share
        </a>
        <div
          className="popover copyPopover popover--l popover--rightAligned popover--bottom">
          <h6
            className="text--left copyPopover__label">
            <i
              className="fa fa-check-circle "
              style={
                Object {
                  "transition": "all 200ms ease-out",
                }
              } />
             Copied to Clipboard!
          </h6>
          <p
            className="copyPopover__copiedString">
            http://localhost/
          </p>
          <input
            className="copyPopover__hidden"
            readOnly={true}
            value="http://localhost/" />
        </div>
      </span>
      <a
        className="actionLink"
        onClick={[Function]}
        style={Object {}}
        title="Refresh Messages">
        Refresh
      </a>
    </div>
    <p
      className="dkimResultListHeader__label marginBottom--xs">
      Generated Test Address
    </p>
    <h3
      className="dkimResultListHeader__email">
      some@email.com
    </h3>
  </div>
</div>
`;

exports[`ResultListHeader component should render with an error present 1`] = `
<div
  className="panel panel--accent">
  <div
    className="panel__body">
    <div
      className="float--right">
      <span
        className="popover__group popover__hoverTrigger">
        <a
          className="actionLink"
          href="https://TEST.sparkpost.com/sign-up?return=http://localhost/"
          onClick={null}
          title="Save Results">
          Save Results
        </a>
        <span
          className="popover popover--top popover--m">
          Create a free SparkPost account or login into your account to save results
        </span>
      </span>
      <span
        className="popover__group"
        onClick={[Function]}>
        <a
          className="actionLink"
          onClick={[Function]}
          style={Object {}}
          title="Share">
          Share
        </a>
        <div
          className="popover copyPopover popover--l popover--rightAligned popover--bottom">
          <h6
            className="text--left copyPopover__label">
            <i
              className="fa fa-check-circle "
              style={
                Object {
                  "transition": "all 200ms ease-out",
                }
              } />
             Copied to Clipboard!
          </h6>
          <p
            className="copyPopover__copiedString">
            http://localhost/
          </p>
          <input
            className="copyPopover__hidden"
            readOnly={true}
            value="http://localhost/" />
        </div>
      </span>
      <a
        className="actionLink"
        onClick={[Function]}
        style={Object {}}
        title="Refresh Messages">
        Refresh
      </a>
    </div>
    <p
      className="dkimResultListHeader__label marginBottom--xs">
      Generated Test Address
    </p>
    <h3
      className="dkimResultListHeader__email">
      some@email.com
    </h3>
  </div>
</div>
`;
